internal/store/builder.go

```diff
@@ -549,6 +584,65 @@ func (b *Builder) buildCustomResourceStores(resourceName string,
 	return stores
 }
 
+func (b *Builder) hasResources(resourceName string, expectedType interface{}) bool {
+	gvr := util.GVRFromType(resourceName, expectedType)
+	if gvr == nil {
+		return true
+	}
+	discoveryClient, err := util.CreateDiscoveryClient(b.utilOptions.Apiserver, b.utilOptions.Kubeconfig)
+	if err != nil {
+		klog.ErrorS(err, "Failed to create discovery client")
+		return false
+	}
+	g := gvr.Group
+	v := gvr.Version
+	r := gvr.Resource
+	isCRDInstalled, err := discovery.IsResourceEnabled(discoveryClient, schema.GroupVersionResource{
+		Group:    g,
+		Version:  v,
+		Resource: r,
+	})
+	if err != nil {
+		klog.ErrorS(err, "Failed to check if CRD is enabled", "group", g, "version", v, "resource", r)
+		return false
+	}
+	if !isCRDInstalled {
+		klog.InfoS("CRD is not installed", "group", g, "version", v, "resource", r)
+		return false
+	}
+	// Wait for the resource to come up.
+	timer := time.NewTimer(10 * time.Second)
```

Comment:
Can we move those durations into a const?

Reply:
Do you mean move this to the `discovery` package as well?